Realm/RLMSyncSubscription.mm

```diff
@@ -0,0 +1,483 @@
+
+#import "RLMSyncSubscription_Private.hpp"
+#import "RLMRealm_Private.hpp"
+#import "RLMUtil.hpp"
+#import "RLMQueryUtil.hpp"
+#import "RLMObjectId_Private.hpp"
+#import "RLMSyncUtil.h"
+
+#import <realm/sync/subscriptions.hpp>
+#import <realm/status_with.hpp>
+#import <realm/util/future.hpp>
+
+#pragma mark - Subscription
+
+@interface RLMSyncSubscription () {
+    std::unique_ptr<realm::sync::Subscription> _subscription;
+    RLMSyncSubscriptionSet *_subscriptionSet;
+}
+@end
+
+@implementation RLMSyncSubscription
+
+- (instancetype)initWithSubscription:(realm::sync::Subscription)subscription subscriptionSet:(RLMSyncSubscriptionSet *)subscriptionSet {
+    if (self = [super init]) {
+        _subscription = std::make_unique<realm::sync::Subscription>(subscription);
+        _subscriptionSet = subscriptionSet;
+        return self;
+    }
+    return nil;
+}
+
+- (RLMObjectId *)identifier {
+    return [[RLMObjectId alloc]initWithValue:_subscription->id()];
+}
+
+- (nullable NSString *)name {
+    const std::string_view str_view = _subscription->name();
+    std::string str = std::string(str_view);
+    const char * characters = str.c_str();
+    NSString *name = [NSString stringWithCString:characters
+                                        encoding:[NSString defaultCStringEncoding]];
+    if ([name length] == 0) {
+        return NULL;
+    }
+    return name;
+}
+
+- (NSDate *)createdAt {
+    return RLMTimestampToNSDate(_subscription->created_at());
+}
+
+- (NSDate *)updatedAt {
+    return RLMTimestampToNSDate(_subscription->updated_at());
+}
+
+- (NSString *)queryString {
+    return [NSString stringWithCString:std::string(_subscription->query_string()).c_str()
+                              encoding:[NSString defaultCStringEncoding]];
+}
+
+- (NSString *)objectClassName {
+    return [NSString stringWithCString:std::string(_subscription->object_class_name()).c_str()
+                              encoding:[NSString defaultCStringEncoding]];
+}
+
+- (void)updateSubscriptionWithClassName:(NSString *)objectClassName
+                                  where:(NSString *)predicateFormat, ... {
+    va_list args;
+    va_start(args, predicateFormat);
+    [self updateSubscriptionWithClassName:objectClassName
+                                    where:predicateFormat
+                                     args:args];
+}
+
+- (void)updateSubscriptionWithClassName:(NSString *)objectClassName
+                                  where:(NSString *)predicateFormat
+                                   args:(va_list)args {
+    [self updateSubscriptionWithClassName:objectClassName
+                                predicate:[NSPredicate predicateWithFormat:predicateFormat arguments:args]];
+    
+}
+
+- (void)updateSubscriptionWithClassName:(NSString *)objectClassName
+                              predicate:(NSPredicate *)predicate {
+    [_subscriptionSet verifyInWriteTransaction];
```

Comment:
Shouldn't the RLMSubscriptionSet method be checking this?

Reply:
cc @dianaafanador3

Reply:
yep you are right, this is already been verified by the `RLMSubscriptionSet`